app/src/main/java/org/apache/taverna/mobile/ui/userprofile/UserProfileFragment.java

```diff
@@ -0,0 +1,154 @@
+
+    @BindView(R.id.user_name)
+    TextView userName;
+
+    @BindView(R.id.user_avatar)
+    CircleImageView userAvatar;
+
+    @BindView(R.id.user_email)
+    TextView userEmail;
+
+    @BindView(R.id.user_website)
+    TextView userWebsite;
+
+    @BindView(R.id.user_description)
+    TextView userDescription;
+
+    @BindView(R.id.user_city)
+    TextView userCity;
+
+    @BindView(R.id.user_country)
+    TextView userCountry;
+
+    private DataManager dataManager;
+
+    public static UserProfileFragment newInstance(String id) {
+
+        Bundle args = new Bundle();
+
+        args.putString(ID, id);
+
+        UserProfileFragment fragment = new UserProfileFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_user_profile, parent, false);
+
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        ButterKnife.bind(this, view);
+
+        setUserDetail();
+
+    }
+
+    @OnClick(R.id.my_workflow_layout)
+    void myWorkflows(View v) {
+        Intent intent = new Intent(getActivity(), MyWorkflowsActivity.class);
+        getActivity().startActivity(intent);
+    }
+
+    @OnClick(R.id.my_favorite_workflow_layout)
+    void myFavoriteWorkflow(View v) {
+        Intent intent = new Intent(getActivity(), FavouriteWorkflowsActivity.class);
+        getActivity().startActivity(intent);
+    }
+
+    private void setUserDetail() {
+
+        dataManager = new DataManager(new PreferencesHelper(getContext()));
+
+        String name = dataManager.getPreferencesHelper().getUserName();
+        String description = dataManager.getPreferencesHelper().getUserDescription();
+        String email = dataManager.getPreferencesHelper().getUserEmail();
+        String website = dataManager.getPreferencesHelper().getUserWebsite();
+        String city = dataManager.getPreferencesHelper().getUserCity();
+        String country = dataManager.getPreferencesHelper().getUserCountry();
+
+        String avatarUrl = dataManager.getPreferencesHelper().getUserAvatar();
+        Glide.with(getContext())
+                .load(avatarUrl)
+                .diskCacheStrategy(DiskCacheStrategy.SOURCE)
+                .placeholder(R.drawable.ic_account_circle_black_24dp)
+                .error(R.drawable.ic_account_circle_black_24dp)
+                .into(new SimpleTarget<GlideDrawable>() {
+                    @Override
+                    public void onResourceReady(GlideDrawable resource, GlideAnimation<? super
+                            GlideDrawable> glideAnimation) {
+                        userAvatar.setImageDrawable(resource);
+                    }
+                });
+
+        if (name != null) {
+            userName.setText(name);
+        } else {
+            userName.setText("<NA>");
```

Comment:
move `<NA>`to string.xml and use it in all fields.

Reply:
@Hiteshgautam01 see this link https://stackoverflow.com/questions/20563524/how-to-add-and-in-android-strings-xml

Reply:
Done